test/webRDFS/WebServerTest.cc

```diff
@@ -42,9 +42,86 @@ TEST(WebServerTest, testRead) {
                    "expectedResultRead actualResultRead"));
 
   system("rm actualResultRead");
-  system("hdfs dfs -fs hdfs://localhost:5351 -rm /fileToRead");
+  system("hdfs dfs -fs hdfs://comp413.local:5351 -rm /fileToRead");
 }
 
+TEST(WebServerTest, testMkdir) {
+  ASSERT_EQ(0,
+            system("curl -i -X PUT "
+                   "https://comp413.local:8080/webhdfs/v1/"
+                   "pathToCreate?op=MKDIRS > actualResultMkdir"));
+
+  // Check that results match
+  ASSERT_EQ(0,
+            system("diff /home/vagrant/rdfs/test/webRDFS/expectedResultMkdir"
+                  " actualResultMkdir"));
+
+  system("rm actualResultMkdir");
+  system("hdfs dfs -fs hdfs://comp413.local:5351 -rm /pathToCreate");
+}
+
+TEST(WebServerTest, testRename) {
+  system("hdfs dfs -fs hdfs://comp413.local:5351 -touchz /fileToRename");
+
+  system("curl -i -X PUT "
+                   "\"https://comp413.local:8080/webhdfs/v1/"
+                   "fileToRename?op=RENAME&destination=newPath\" > "
+                   "actualResultRename");
+
+  // Check that results match
+  ASSERT_EQ(0,
+            system("diff /home/vagrant/rdfs/test/webRDFS/expectedResultRename"
+                   " actualResultRename"));
+
+  system("rm actualResultRename");
+  system("hdfs dfs -fs hdfs://comp413.local:5351 -rm /fileToRename");
+}
+
+TEST(WebServerTest, testFrontend) {
+  ASSERT_EQ(0, system("curl -i https://comp413.local:8080 > frontend"));
+  ASSERT_EQ(0, system("grep \"Content-Type: text/html\" frontend"));
+  ASSERT_EQ(0, system("rm frontend"));
+}
+
+TEST(WebServerTest, testCreate) {
+  system(
+          "hdfs dfs -fs hdfs://comp413.local:5351 "
+                  "-copyFromLocal fileForTesting /fileToCreate");
+
+  system("curl -i -X PUT "
+                   "\"https://comp413.local:8080/webhdfs/v1/"
+                   "fileToCreate?op=CREATE\" > actualResultCreate");
+
+  // Check that results match
+  ASSERT_EQ(0,
+            system("diff /home/vagrant/rdfs/test/webRDFS/"
+                           "expectedResultCreate actualResultCreate"));
+
+  system("rm actualResultCreate");
+  system("hdfs dfs -fs hdfs://comp413.local:5351 -rm /fileToCreate");
+}
+
+TEST(WebServerTest, testListing) {
+  system(
+          "hdfs dfs -fs hdfs://comp413.local:5351 "
+                  "-mkdir /dirToLs");
+  system(
+          "hdfs dfs -fs hdfs://comp413.local:5351 "
+          "-copyFromLocal /home/vagrant/rdfs/test/webRDFS/fileForTesting "
+           "/dirToLs/");
+
+  ASSERT_EQ(0,
+            system("curl -i https://comp413.local:8080/webhdfs/v1/"
+                           "dirToLs?op=LISTSTATUS > actualResultLs"));
+
+  // Check that results match except for the access times
+  ASSERT_EQ(256,
```

Comment:
This seems strange, why do you assert that the expected and actual results are completely different?